src/GoTrueClient.ts

```diff
@@ -640,9 +564,13 @@ export default class GoTrueClient {
    * For server-side management, you can revoke all refresh tokens for a user by passing a user's JWT through to `auth.api.signOut(JWT: string)`. There is no way to revoke a user's session JWT before it automatically expires
    */
   async signOut(): Promise<{ error: AuthError | null }> {
-    const accessToken = this.currentSession?.access_token
+    const { session, error: sessionError } = await this.getSession()
+    if (sessionError) {
+      throw sessionError
```

Comment:
Could we return instead of throwing here? `return {error: sessionError}` There is no try/catch in the signOut method, and sessionError should always return an `AuthError`.